edm4eic.yaml

```diff
@@ -135,6 +135,27 @@ components:
         }\n
       "
 
+  edm4eic::Cov6f:
+    Members:
+      - std::array<float, 21> covariance  // 6d triangular packed covariance matrix
+    ExtraCode:
+      declaration: "
+        Cov6f() : covariance{} {}\n
+        Cov6f(std::array<float, 21> vcov) : covariance{vcov}{}\n
+	float getCov(unsigned i, unsigned j) const {\n
+	  if(i > j) {\n
+     	    std::swap(i, j);\n
+	  }\n
+	  return covariance[i + 1 + (j + 1) * (j) / 2 - 1];\n
+	}\n
+	void setCov(unsigned i, unsigned j, float val) {\n
```

Comment:
This kinda duplicates the same serialization code. Is it possible to have just one that returns a reference, `float& cov(unsigned i, unsigned j)` and then usage is `cov(i,j) = 0.1`?

Reply:
No (well, in part; consistency with other Covs is desireable too), I meant returning a reference to the underlying quantity that can then be used in assignment as well, e.g. what you likely know from Acts (e.g. https://github.com/acts-project/acts/blob/1f6dd58d096f89452c1f5d8d5da65b6ddbb7e0bd/Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSmearing.cpp#L138), which comes from Eigen at https://eigen.tuxfamily.org/dox/classEigen_1_1DenseCoeffsBase_3_01Derived_00_01WriteAccessors_01_4.html#ae6ba07bad9e3026afe54806fdefe32bb. I think that requires non-const function with return type reference, and can be operator() or a named function.

Reply:
Yeah I think I updated it to what you meant, let me know if that is not true (minus the return by reference, added in my local version)